### PR TITLE
mkvtoolnix: 8.4.0 -> 8.9.0

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Things done:
- [x] Tested using sandboxing
- [x] Built on platform(s): NixOS
- [x] Tested compilation of all pkgs that depend on this change using: none
- [x] Tested execution of all binary files
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

Fixes issue #13714

cc @Fuuzetsu @codyopel

### CODE REVIEW

File: pkgs/applications/video/mkvtoolnix/default.nix [hunk 1]
```diff
@@ -1,76 +1,54 @@
...
+    pkgconfig autoconf automake expat
+    file xdg_utils boost libebml zlib
+    libmatroska libogg libvorbis flac
+    (optional withGUI qt5.qtbase)
+  ];
 
-  buildPhase = ''
-    ./drake
-  '';
+  preConfigure = "./autogen.sh";
```
Comment (globin) [inline pkgs/applications/video/mkvtoolnix/default.nix]: using the autoreconfHook in buildInputs is preferred, see: https://github.com/NixOS/nixpkgs/commit/495985ace397cb24f6d3bc317945aa32fdabf4db

Comment (rnhmjoj) [inline pkgs/applications/video/mkvtoolnix/default.nix]: Oh thanks, I was just wondering about this.

Comment (rnhmjoj) [inline pkgs/applications/video/mkvtoolnix/default.nix]: Uhm, it doesn't work. I get this:

    configure: error: cannot run /nix/store/idm1067y9i6m87crjqrbamdsq2ma5r7p-bash-4.3-p42/bin/bash ./config.sub

**Comment by aszlig** [comment]
@rnhmjoj: Using nix-build with `--option build-use-chroot` won't work here and it will build *without* chroot. So you either have to set `build-use-chroot` in your `nix.conf` or not use the daemon *at all* and write directly to the store.

**Comment by rnhmjoj** [comment]
@aszlig So setting nix.useChroot will do?

**Comment by aszlig** [comment]
@rnhmjoj: Yup

**Comment by rnhmjoj** [comment]
Is there something better than this to patch the shebangs?
```sh
    files=$(grep -rl '/usr/bin/env ruby')
    for file in $files; do
      substituteInPlace $file --replace '/usr/bin/env ruby' '${ruby}/bin/ruby'
    done
```

**Comment by joachifm** [comment]
`patchShebangs`?

**Pull request merged**